<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Applying PROGENy on single-cell RNA-seq data • progeny</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Applying PROGENy on single-cell RNA-seq data">
<meta property="og:description" content="progeny">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-8"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-8');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">progeny</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ProgenySingleCell.html">Applying PROGENy on single-cell RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/progenyBulk.html">PROGENy pathway signatures: Application to Bulk transcriptomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/progeny/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Applying PROGENy on single-cell RNA-seq data</h1>
                        <h4 class="author">Alberto Valdeolivas</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:alberto.valdeolivas@bioquant.uni-heidelberg.de" class="email">alberto.valdeolivas@bioquant.uni-heidelberg.de</a>
      </address>
                              <h4 class="author">Igor Bulanov</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><h4 class="author">Christian Holland</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><h4 class="author">Julio Saez-Rodriguez</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/progeny/blob/master/vignettes/ProgenySingleCell.Rmd"><code>vignettes/ProgenySingleCell.Rmd</code></a></small>
      <div class="hidden name"><code>ProgenySingleCell.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to infer the activity of 14 relevant signaling pathways by running PROGENy on single-cell RNA-seq data.</p>
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Conventional pathway analysis methods rely on the gene expression of the pathway members. However, this approach overlook the effect of post-translational modifications and only captures very specific experimental conditions. To overcome these limitations, <strong>PROGENy</strong> (Pathway RespOnsive GENes) estimates the activity of relevant signaling pathways based on consensus gene signatures obtained from perturbation experiments, in other words, the footprint of the pathway on gene expression <span class="citation">(Schubert et al. 2018 ,<span class="citation">@dugourd2019</span>)</span>.</p>
<p><strong>PROGENy</strong> initially contained 11 pathways and was developed for the application to human transcriptomics data. It has been recently shown that <strong>PROGENy</strong> is also applicable to mouse data <span class="citation">(Holland, Szalai, and Saez-Rodriguez 2019)</span> and to single cell RNAseq data <span class="citation">(Holland et al. 2019)</span>. In addition, they expanded human and mouse <strong>PROGENy</strong> to 14 pathways.</p>
<p>This vignette shows an example on how to apply <strong>PROGENy</strong> in a well known single-cell dataset. To analyse the data, we followed the standard procedures of the <em><a href="https://CRAN.R-project.org/package=Seurat">Seurat</a></em> toolkit for single cell genomics <span class="citation">(Stuart et al. 2019)</span>.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>First of all, you need a current version of R (<a href="http://www.r-project.org" class="uri">http://www.r-project.org</a>). <em><a href="https://bioconductor.org/packages/3.11/progeny">progeny</a></em> is a freely available annotation package deposited on <a href="http://bioconductor.org/" class="uri">http://bioconductor.org/</a> and <a href="https://https://github.com/saezlab/progeny" class="uri">https://https://github.com/saezlab/progeny</a>.</p>
<p>You can install it by running the following commands on an R console:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"progeny"</span>)

<span class="co">## To install the new version until it is submited to Bioconductor use:</span>
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"saezlab/progeny"</span>)</pre></body></html></div>
<p>We also load here the packages required to run this script.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">progeny</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Seurat</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tidyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">readr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">pheatmap</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tibble</span>)</pre></body></html></div>
</div>
<div id="example-of-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#example-of-usage" class="anchor"></a>Example of usage</h1>
<p>In the following paragraphs, we provide examples describing how to run <strong>PROGENy</strong> on a scRNA-seq dataset. In particular, we use the <em><a href="https://CRAN.R-project.org/package=Seurat">Seurat</a></em> toolkit for single cell genomics <span class="citation">(Stuart et al. 2019)</span>. For the sake of simplicity, we follow the example provided in the following <em><a href="https://CRAN.R-project.org/package=Seurat">Seurat</a></em> vignette:</p>
<p><a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html" class="uri">https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html</a></p>
<p>The dataset contains 2,700 Peripheral Blood Mononuclear Cells (PBMC) that were sequenced on the Illumina NextSeq 500. This dataset is freely available in 10X Genomics:</p>
<p><a href="https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz" class="uri">https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz</a></p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co">## Load the PBMC dataset</span>
<span class="no">pbmc.data</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Read10X.html">Read10X</a></span>(<span class="kw">data.dir</span> <span class="kw">=</span> <span class="st">"../data/pbmc3k/filtered_gene_bc_matrices/hg19/"</span>)
<span class="co">## Initialize the Seurat object with the raw (non-normalized data).</span>
<span class="no">pbmc</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/CreateSeuratObject.html">CreateSeuratObject</a></span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="no">pbmc.data</span>, <span class="kw">project</span> <span class="kw">=</span> <span class="st">"pbmc3k"</span>, <span class="kw">min.cells</span> <span class="kw">=</span> <span class="fl">3</span>,
    <span class="kw">min.features</span> <span class="kw">=</span> <span class="fl">200</span>)</pre></body></html></div>
<div id="pre-processing-normalization-and-identification-of-highly-variable-features" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-processing-normalization-and-identification-of-highly-variable-features" class="anchor"></a>Pre-processing, normalization and identification of highly variable features</h2>
<p>We follow the standard pre-processing steps as described in the aforementioned <em><a href="https://CRAN.R-project.org/package=Seurat">Seurat</a></em> vignette before going deeper into the data analysis. These steps carry out the selection and filtration of cells based on quality control metrics, the data normalization and scaling, and the detection of highly variable features (see <a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html" class="uri">https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html</a>).</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co">## Identification of mithocondrial genes</span>
<span class="no">pbmc</span><span class="kw">[[</span><span class="st">"percent.mt"</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/PercentageFeatureSet.html">PercentageFeatureSet</a></span>(<span class="no">pbmc</span>, <span class="kw">pattern</span> <span class="kw">=</span> <span class="st">"^MT-"</span>)

<span class="co">## Filtering cells following standard QC criteria.</span>
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/subset.Seurat.html">subset</a></span>(<span class="no">pbmc</span>, <span class="kw">subset</span> <span class="kw">=</span> <span class="no">nFeature_RNA</span> <span class="kw">&gt;</span> <span class="fl">200</span> <span class="kw">&amp;</span> <span class="no">nFeature_RNA</span> <span class="kw">&lt;</span> <span class="fl">2500</span> <span class="kw">&amp;</span>
    <span class="no">percent.mt</span> <span class="kw">&lt;</span> <span class="fl">5</span>)

<span class="co">## Normalizing the data</span>
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/NormalizeData.html">NormalizeData</a></span>(<span class="no">pbmc</span>, <span class="kw">normalization.method</span> <span class="kw">=</span> <span class="st">"LogNormalize"</span>,
    <span class="kw">scale.factor</span> <span class="kw">=</span> <span class="fl">10000</span>)

<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/NormalizeData.html">NormalizeData</a></span>(<span class="no">pbmc</span>)

<span class="co">## Identify the 2000 most highly variable genes</span>
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/FindVariableFeatures.html">FindVariableFeatures</a></span>(<span class="no">pbmc</span>, <span class="kw">selection.method</span> <span class="kw">=</span> <span class="st">"vst"</span>, <span class="kw">nfeatures</span> <span class="kw">=</span> <span class="fl">2000</span>)

<span class="co">## In addition we scale the data</span>
<span class="no">all.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames</a></span>(<span class="no">pbmc</span>)
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/ScaleData.html">ScaleData</a></span>(<span class="no">pbmc</span>, <span class="kw">features</span> <span class="kw">=</span> <span class="no">all.genes</span>)</pre></body></html></div>
</div>
<div id="clustering-cells" class="section level2">
<h2 class="hasAnchor">
<a href="#clustering-cells" class="anchor"></a>Clustering cells</h2>
<p>One of the most relevant steps in scRNA-seq data analysis is clustering. Cells are grouped based on the similarity of their transcriptomic profiles. We first apply the <em><a href="https://CRAN.R-project.org/package=Seurat">Seurat</a></em> v3 classical approach as described in their aforementioned vignette. We visualize the cell clusters using UMAP:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">pbmc</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/RunPCA.html">RunPCA</a></span>(<span class="no">pbmc</span>, <span class="kw">features</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/VariableFeatures.html">VariableFeatures</a></span>(<span class="kw">object</span> <span class="kw">=</span> <span class="no">pbmc</span>), <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/FindNeighbors.html">FindNeighbors</a></span>(<span class="no">pbmc</span>, <span class="kw">dims</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fl">10</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/FindClusters.html">FindClusters</a></span>(<span class="no">pbmc</span>, <span class="kw">resolution</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/RunUMAP.html">RunUMAP</a></span>(<span class="no">pbmc</span>, <span class="kw">dims</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fl">10</span>,  <span class="kw">umap.method</span> <span class="kw">=</span> <span class="st">'uwot'</span>, <span class="kw">metric</span><span class="kw">=</span><span class="st">'cosine'</span>)</pre></body></html></div>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/DimPlot.html">DimPlot</a></span>(<span class="no">pbmc</span>, <span class="kw">reduction</span> <span class="kw">=</span> <span class="st">"umap"</span>)</pre></body></html></div>
<p><img src="ProgenySingleCell_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div id="pathway-activity-per-cell-population" class="section level2">
<h2 class="hasAnchor">
<a href="#pathway-activity-per-cell-population" class="anchor"></a>Pathway activity per cell population</h2>
<p>Following again <em><a href="https://CRAN.R-project.org/package=Seurat">Seurat</a></em> protocol, we next find the markers that help to identify and discriminate the different cell popualations present in the dataset under study.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## Finding differentially expressed features (cluster biomarkers)</span>
<span class="no">pbmc.markers</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/FindAllMarkers.html">FindAllMarkers</a></span>(<span class="no">pbmc</span>, <span class="kw">only.pos</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.25</span>,
    <span class="kw">logfc.threshold</span> <span class="kw">=</span> <span class="fl">0.25</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="co">## Assigning cell type identity to clusters</span>
<span class="no">new.cluster.ids</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Naive CD4 T"</span>, <span class="st">"Memory CD4 T"</span>, <span class="st">"CD14+ Mono"</span>, <span class="st">"B"</span>, <span class="st">"CD8 T"</span>,
    <span class="st">"FCGR3A+ Mono"</span>, <span class="st">"NK"</span>, <span class="st">"DC"</span>, <span class="st">"Platelet"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">new.cluster.ids</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="no">pbmc</span>)
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">RenameIdents</a></span>(<span class="no">pbmc</span>, <span class="no">new.cluster.ids</span>)

<span class="co">## We create a data frame with the specification of the cells that belong to </span>
<span class="co">## each cluster to match with the Progeny scores.</span>
<span class="no">CellsClusters</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">Cell</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">pbmc</span>)),
    <span class="kw">CellType</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">pbmc</span>)),
    <span class="kw">stringsAsFactors</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p>We plot again the clusters along with their cell type labels</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/DimPlot.html">DimPlot</a></span>(<span class="no">pbmc</span>, <span class="kw">reduction</span> <span class="kw">=</span> <span class="st">"umap"</span>, <span class="kw">label</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">pt.size</span> <span class="kw">=</span> <span class="fl">0.5</span>) + <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/SeuratTheme.html">NoLegend</a></span>()</pre></body></html></div>
<p><img src="ProgenySingleCell_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>Finally, we compute <strong>PROGENy</strong> pathway activity scores on the scRNA-seq data, and we then characterice the different cell populations based on these scores.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">## We compute the Progeny activity scores and add them to our Seurat object</span>
<span class="co">## as a new assay called Progeny. </span>
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/progeny.html">progeny</a></span>(<span class="no">pbmc</span>, <span class="kw">scale</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">organism</span><span class="kw">=</span><span class="st">"Human"</span>, <span class="kw">top</span><span class="kw">=</span><span class="fl">500</span>, <span class="kw">perm</span><span class="kw">=</span><span class="fl">1</span>,
    <span class="kw">return_assay</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co">## We can now directly apply Seurat functions in our Progeny scores. </span>
<span class="co">## For instance, we scale the pathway activity scores. </span>
<span class="no">pbmc</span> <span class="kw">&lt;-</span> <span class="kw pkg">Seurat</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/ScaleData.html">ScaleData</a></span>(<span class="no">pbmc</span>, <span class="kw">assay</span> <span class="kw">=</span> <span class="st">"progeny"</span>)

<span class="co">## We transform Progeny scores into a data frame to better handling the results</span>
<span class="no">progeny_scores_df</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/GetAssayData.html">GetAssayData</a></span>(<span class="no">pbmc</span>, <span class="kw">slot</span> <span class="kw">=</span> <span class="st">"scale.data"</span>,
        <span class="kw">assay</span> <span class="kw">=</span> <span class="st">"progeny"</span>))) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span>(<span class="st">"Cell"</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/gather.html">gather</a></span>(<span class="no">Pathway</span>, <span class="no">Activity</span>, -<span class="no">Cell</span>)

<span class="co">## We match Progeny scores with the cell clusters.</span>
<span class="no">progeny_scores_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span>(<span class="no">progeny_scores_df</span>, <span class="no">CellsClusters</span>)

<span class="co">## We summarize the Progeny scores by cellpopulation</span>
<span class="no">summarized_progeny_scores</span> <span class="kw">&lt;-</span> <span class="no">progeny_scores_df</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="no">Pathway</span>, <span class="no">CellType</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="kw">avg</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="no">Activity</span>), <span class="kw">std</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span>(<span class="no">Activity</span>))</pre></body></html></div>
<p>We plot the different pathway activities for the different cell populations</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co">## We prepare the data for the plot</span>
<span class="no">summarized_progeny_scores_df</span> <span class="kw">&lt;-</span> <span class="no">summarized_progeny_scores</span> <span class="kw">%&gt;%</span>
    <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(-<span class="no">std</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/spread.html">spread</a></span>(<span class="no">Pathway</span>, <span class="no">avg</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">check.names</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">stringsAsFactors</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">paletteLength</span> <span class="kw">=</span> <span class="fl">100</span>
<span class="no">myColor</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Darkblue"</span>, <span class="st">"white"</span>,<span class="st">"red"</span>))(<span class="no">paletteLength</span>)

<span class="no">progenyBreaks</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="no">summarized_progeny_scores_df</span>), <span class="fl">0</span>,
                      <span class="kw">length.out</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">ceiling</a></span>(<span class="no">paletteLength</span>/<span class="fl">2</span>) + <span class="fl">1</span>),
                  <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="no">summarized_progeny_scores_df</span>)/<span class="no">paletteLength</span>,
                      <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="no">summarized_progeny_scores_df</span>),
                      <span class="kw">length.out</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">floor</a></span>(<span class="no">paletteLength</span>/<span class="fl">2</span>)))
<span class="no">progeny_hmap</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="no">summarized_progeny_scores_df</span>[,-<span class="fl">1</span>]),<span class="kw">fontsize</span><span class="kw">=</span><span class="fl">14</span>,
                        <span class="kw">fontsize_row</span> <span class="kw">=</span> <span class="fl">10</span>,
                        <span class="kw">color</span><span class="kw">=</span><span class="no">myColor</span>, <span class="kw">breaks</span> <span class="kw">=</span> <span class="no">progenyBreaks</span>,
                        <span class="kw">main</span> <span class="kw">=</span> <span class="st">"PROGENy (500)"</span>, <span class="kw">angle_col</span> <span class="kw">=</span> <span class="fl">45</span>,
                        <span class="kw">treeheight_col</span> <span class="kw">=</span> <span class="fl">0</span>,  <span class="kw">border_color</span> <span class="kw">=</span> <span class="fl">NA</span>)</pre></body></html></div>
<p><img src="ProgenySingleCell_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 4.0.0 (2020-04-24)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 19.10
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.3.7.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] tibble_3.0.1     pheatmap_1.0.12  readr_1.3.1      tidyr_1.1.0     
## [5] ggplot2_3.3.1    Seurat_3.1.5     dplyr_1.0.0      progeny_1.11.1  
## [9] BiocStyle_2.16.0
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-147        tsne_0.1-3          fs_1.4.1           
##  [4] RcppAnnoy_0.0.16    RColorBrewer_1.1-2  httr_1.4.1         
##  [7] rprojroot_1.3-2     sctransform_0.2.1   tools_4.0.0        
## [10] backports_1.1.7     R6_2.4.1            irlba_2.3.3        
## [13] KernSmooth_2.23-17  uwot_0.1.8          lazyeval_0.2.2     
## [16] colorspace_1.4-1    withr_2.2.0         tidyselect_1.1.0   
## [19] gridExtra_2.3       compiler_4.0.0      desc_1.2.0         
## [22] plotly_4.9.2.1      labeling_0.3        bookdown_0.19      
## [25] scales_1.1.1        lmtest_0.9-37       ggridges_0.5.2     
## [28] pbapply_1.4-2       rappdirs_0.3.1      pkgdown_1.5.1      
## [31] stringr_1.4.0       digest_0.6.25       rmarkdown_2.2      
## [34] pkgconfig_2.0.3     htmltools_0.4.0     limma_3.44.1       
## [37] htmlwidgets_1.5.1   rlang_0.4.6         rstudioapi_0.11    
## [40] farver_2.0.3        generics_0.0.2      zoo_1.8-8          
## [43] jsonlite_1.6.1      ica_1.0-2           magrittr_1.5       
## [46] patchwork_1.0.0     Matrix_1.2-18       Rcpp_1.0.4.6       
## [49] munsell_0.5.0       ape_5.4             reticulate_1.16    
## [52] lifecycle_0.2.0     stringi_1.4.6       yaml_2.2.1         
## [55] MASS_7.3-51.6       Rtsne_0.15          plyr_1.8.6         
## [58] grid_4.0.0          parallel_4.0.0      listenv_0.8.0      
## [61] ggrepel_0.8.2       crayon_1.3.4        lattice_0.20-41    
## [64] cowplot_1.0.0       splines_4.0.0       hms_0.5.3          
## [67] knitr_1.28          pillar_1.4.4        igraph_1.2.5       
## [70] future.apply_1.5.0  reshape2_1.4.4      codetools_0.2-16   
## [73] leiden_0.3.3        glue_1.4.1          evaluate_0.14      
## [76] data.table_1.12.8   BiocManager_1.30.10 vctrs_0.3.1        
## [79] png_0.1-7           gtable_0.3.0        RANN_2.6.1         
## [82] purrr_0.3.4         future_1.17.0       assertthat_0.2.1   
## [85] xfun_0.14           rsvd_1.0.3          RSpectra_0.16-0    
## [88] survival_3.1-12     viridisLite_0.3.0   memoise_1.1.0      
## [91] cluster_2.1.0       globals_0.12.5      fitdistrplus_1.1-1 
## [94] ellipsis_0.3.1      ROCR_1.0-11</code></pre>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-dugourd2019">
<p>Dugourd, Aurelien, and Julio Saez-Rodriguez. 2019. “Footprint-Based Functional Analysis of Multiomic Data.” <em>Current Opinion in Systems Biology</em> 15: 82–90. <a href="https://doi.org/10.1016/j.coisb.2019.04.002">https://doi.org/10.1016/j.coisb.2019.04.002</a>.</p>
</div>
<div id="ref-HOLLAND2019194431">
<p>Holland, Christian H., Bence Szalai, and Julio Saez-Rodriguez. 2019. “Transfer of Regulatory Knowledge from Human to Mouse for Functional Genomics Analysis.” <em>Biochimica et Biophysica Acta (BBA) - Gene Regulatory Mechanisms</em>, 194431. <a href="https://doi.org/https://doi.org/10.1016/j.bbagrm.2019.194431">https://doi.org/https://doi.org/10.1016/j.bbagrm.2019.194431</a>.</p>
</div>
<div id="ref-Holland2019">
<p>Holland, Christian H., Jovan Tanevski, Jan Gleixner, Manu P. Kumar, Elisabetta Mereu, Javier Perales-Patón, Brian A. Joughin, et al. 2019. “Robustness and Applicability of Functional Genomics Tools on scRNA-Seq Data.” <em>bioRxiv</em>. <a href="https://doi.org/10.1101/753319">https://doi.org/10.1101/753319</a>.</p>
</div>
<div id="ref-Schubert2018">
<p>Schubert, Michael, Bertram Klinger, Martina Klünemann, Anja Sieber, Florian Uhlitz, Sascha Sauer, Mathew J. Garnett, Nils Blüthgen, and Julio Saez-Rodriguez. 2018. “Perturbation-Response Genes Reveal Signaling Footprints in Cancer Gene Expression.” <em>Nature Communications</em> 9 (1): 20. <a href="https://doi.org/10.1038/s41467-017-02391-6">https://doi.org/10.1038/s41467-017-02391-6</a>.</p>
</div>
<div id="ref-Stuart2019">
<p>Stuart, Tim, Andrew Butler, Paul Hoffman, Christoph Hafemeister, Efthymia Papalexi, William M. III Mauck, Yuhan Hao, Marlon Stoeckius, Peter Smibert, and Rahul Satija. 2019. “Comprehensive Integration of Single-Cell Data.” <em>Cell</em> 177 (7): 1888–1902.e21. <a href="https://doi.org/10.1016/j.cell.2019.05.031">https://doi.org/10.1016/j.cell.2019.05.031</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Michael Schubert, Alberto Valdeolivas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
