<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PROGENy pathway signatures: Application to Bulk transcriptomics • progeny</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="PROGENy pathway signatures: Application to Bulk transcriptomics">
<meta property="og:description" content="progeny">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-8"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-8');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">progeny</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ProgenySingleCell.html">Applying PROGENy on single-cell RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/progenyBulk.html">PROGENy pathway signatures: Application to Bulk transcriptomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/progeny/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="progenyBulk_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PROGENy pathway signatures: Application to Bulk transcriptomics</h1>
                        <h4 class="author">Michael Schubert</h4>
            
            <h4 class="date">2020-09-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/progeny/blob/master/vignettes/progenyBulk.Rmd"><code>vignettes/progenyBulk.Rmd</code></a></small>
      <div class="hidden name"><code>progenyBulk.Rmd</code></div>

    </div>

    
    
<div id="progeny-pathway-signatures" class="section level1">
<h1 class="hasAnchor">
<a href="#progeny-pathway-signatures" class="anchor"></a>PROGENy pathway signatures</h1>
<p>This R package provides the model we inferred in the publication “Perturbation-response genes reveal signaling footprints in cancer gene expression” and a function to obtain pathway scores from a gene expression matrix. It is <a href="http://www.biorxiv.org/content/early/2016/08/28/065672">available on bioRxiv</a>.</p>
<div id="scoring-the-airway-package-data-for-pathway-scores" class="section level2">
<h2 class="hasAnchor">
<a href="#scoring-the-airway-package-data-for-pathway-scores" class="anchor"></a>Scoring the <code>airway</code> package data for pathway scores</h2>
<p>This is to outline how to prepare expression data, in this case from the <code>airway</code> package for pathway activity analysis using PROGENy.</p>
<div id="preparing-the-gene-expression-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#preparing-the-gene-expression-matrix" class="anchor"></a>Preparing the gene expression matrix</h3>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">airway</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">DESeq2</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">airway</span>)

<span class="co"># import data to DESeq2 and variance stabilize</span>
<span class="no">dset</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span>(<span class="fu">assay</span>(<span class="no">airway</span>),
    <span class="kw">colData</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">colData</span>(<span class="no">airway</span>)), <span class="kw">design</span><span class="kw">=</span>~<span class="no">dex</span>)
<span class="no">dset</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/estimateSizeFactors.html">estimateSizeFactors</a></span>(<span class="no">dset</span>)
<span class="no">dset</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/estimateDispersions.html">estimateDispersions</a></span>(<span class="no">dset</span>)
<span class="no">gene_expr</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/varianceStabilizingTransformation.html">getVarianceStabilizedData</a></span>(<span class="no">dset</span>)

<span class="co"># annotate matrix with HGNC symbols</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">biomaRt</span>)
<span class="no">mart</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useDataset.html">useDataset</a></span>(<span class="st">"hsapiens_gene_ensembl"</span>, <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span>(<span class="st">"ensembl"</span>))
<span class="no">genes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html">getBM</a></span>(<span class="kw">attributes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ensembl_gene_id"</span>,<span class="st">"hgnc_symbol"</span>),
              <span class="kw">values</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gene_expr</span>), <span class="kw">mart</span><span class="kw">=</span><span class="no">mart</span>)
<span class="no">matched</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gene_expr</span>), <span class="no">genes</span>$<span class="no">ensembl_gene_id</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gene_expr</span>) <span class="kw">=</span> <span class="no">genes</span>$<span class="no">hgnc_symbol</span>[<span class="no">matched</span>]</pre></body></html></div>
</div>
<div id="obtaining-pathway-scores" class="section level3">
<h3 class="hasAnchor">
<a href="#obtaining-pathway-scores" class="anchor"></a>Obtaining pathway scores</h3>
<p>We can then use the <code>progeny</code> function to score the expression matrix. Note that we are scaling the pathway scores with respect to the controls only.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">progeny</span>)
<span class="no">pathways</span> <span class="kw">=</span> <span class="fu"><a href="../reference/progeny.html">progeny</a></span>(<span class="no">gene_expr</span>, <span class="kw">scale</span><span class="kw">=</span><span class="fl">FALSE</span>)
<span class="no">controls</span> <span class="kw">=</span> <span class="no">airway</span>$<span class="no">dex</span> <span class="kw">==</span> <span class="st">"untrt"</span>
<span class="no">ctl_mean</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">pathways</span>[<span class="no">controls</span>,], <span class="fl">2</span>, <span class="no">mean</span>)
<span class="no">ctl_sd</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">pathways</span>[<span class="no">controls</span>,], <span class="fl">2</span>, <span class="no">sd</span>)
<span class="no">pathways</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">pathways</span>, <span class="fl">1</span>, <span class="kw">function</span>(<span class="no">x</span>) <span class="no">x</span> - <span class="no">ctl_mean</span>))
<span class="no">pathways</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">pathways</span>, <span class="fl">1</span>, <span class="kw">function</span>(<span class="no">x</span>) <span class="no">x</span> / <span class="no">ctl_sd</span>)</pre></body></html></div>
</div>
<div id="checking-for-differences-between-the-groups" class="section level3">
<h3 class="hasAnchor">
<a href="#checking-for-differences-between-the-groups" class="anchor"></a>Checking for differences between the groups</h3>
<p>So now we might be interested how the treatment with dexamethasone affects signaling pathways. To do this, we check if the control is different to the perturbed condition using a linear model:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="no">result</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">pathways</span>, <span class="fl">1</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="kw pkg">broom</span><span class="kw ns">::</span><span class="fu"><a href="https://broom.tidymodels.org/reference/reexports.html">tidy</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span>(<span class="no">x</span> ~ !<span class="no">controls</span>)) <span class="kw">%&gt;%</span>
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">term</span> <span class="kw">==</span> <span class="st">"!controlsTRUE"</span>) <span class="kw">%&gt;%</span>
        <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(-<span class="no">term</span>)
})
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(<span class="no">result</span>), <span class="kw">pathway</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">result</span>))</pre></body></html></div>
<pre><code>## # A tibble: 14 x 5
##    estimate std.error statistic   p.value pathway 
##       &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;   
##  1    8.89      0.830    10.7   0.0000392 Androgen
##  2    1.25      3.22      0.389 0.711     EGFR    
##  3   -6.90      0.830    -8.31  0.000165  Estrogen
##  4    0.783     1.92      0.408 0.698     Hypoxia 
##  5   -0.972     0.657    -1.48  0.190     JAK-STAT
##  6    1.04      1.34      0.774 0.469     MAPK    
##  7    0.456     0.697     0.655 0.537     NFkB    
##  8   -4.42      0.787    -5.61  0.00137   p53     
##  9   -0.618     1.94     -0.319 0.761     PI3K    
## 10    2.34      1.09      2.15  0.0756    TGFb    
## 11    0.715     0.681     1.05  0.335     TNFa    
## 12    0.279     0.785     0.356 0.734     Trail   
## 13    0.703     0.705     0.996 0.358     VEGF    
## 14   -1.75      0.566    -3.10  0.0211    WNT</code></pre>
<p>What we see is that indeed the p53/DNA damage response pathway is less active after treatment than before.</p>
</div>
</div>
<div id="reproducing-drug-associations-on-the-gdsc-panel" class="section level2">
<h2 class="hasAnchor">
<a href="#reproducing-drug-associations-on-the-gdsc-panel" class="anchor"></a>Reproducing drug associations on the GDSC panel</h2>
<p>Below is an example on how to calculate pathway scores for cell lines in the Genomics of Drug Sensitivity in Cancer (GDSC) panel, and to check for associations with drug response.</p>
<p>The code used for the analyses is <a href="https://github.com/saezlab/footprints">available on Github</a>.</p>
<div id="getting-the-data" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-the-data" class="anchor"></a>Getting the data</h3>
<p>This example shows how to use the GDSC gene expression data of multiple cell lines together with PROGENy to calculate pathway activity and then to check for associations with drug sensitivity.</p>
<p>First, we need the GDSC data for both gene expression and drug response. They are available on the <a href="http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Home.html">GDSC1000 web site</a>:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co"># set up a file cache so we download only once</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BiocFileCache</span>)
<span class="no">bfc</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html">BiocFileCache</a></span>(<span class="st">"."</span>)
<span class="co"># gene expression and drug response</span>
<span class="no">base</span> <span class="kw">=</span> <span class="st">"http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Data/"</span>
<span class="no">paths</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html">bfcrpath</a></span>(<span class="no">bfc</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">base</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"suppData/TableS4A.xlsx"</span>,
            <span class="st">"preprocessed/Cell_line_RMA_proc_basalExp.txt.zip"</span>)))</pre></body></html></div>
<p>You can also download the files manually (adjust the file names when loading):</p>
<ul>
<li><a href="http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Data/preprocessed/Cell_line_RMA_proc_basalExp.txt.zip">Processed gene expression matrix</a></li>
<li><a href="http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Data/suppData/TableS4A.xlsx">Drug sensitivities</a></li>
</ul>
</div>
<div id="creating-the-right-objects-to-work-with" class="section level3">
<h3 class="hasAnchor">
<a href="#creating-the-right-objects-to-work-with" class="anchor"></a>Creating the right objects to work with</h3>
<p>First, we need to load the files we just downloaded into R to be able to perform the analysis:</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co"># load the downloaded files</span>
<span class="no">drug_table</span> <span class="kw">&lt;-</span> <span class="kw pkg">readxl</span><span class="kw ns">::</span><span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html">read_excel</a></span>(<span class="no">paths</span>[<span class="fl">1</span>], <span class="kw">skip</span><span class="kw">=</span><span class="fl">5</span>, <span class="kw">na</span><span class="kw">=</span><span class="st">"NA"</span>)
<span class="no">drug_table</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/replace.html">replace</a></span>(<span class="no">drug_table</span>, <span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">drug_table</span>), <span class="fl">0</span>)
<span class="no">gene_table</span> <span class="kw">&lt;-</span> <span class="kw pkg">readr</span><span class="kw ns">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span>(<span class="no">paths</span>[<span class="fl">2</span>])

<span class="co"># we need drug response with COSMIC IDs</span>
<span class="no">drug_response</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html">data.matrix</a></span>(<span class="no">drug_table</span>[,<span class="fl">3</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">drug_table</span>)])
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">drug_response</span>) <span class="kw">&lt;-</span> <span class="no">drug_table</span><span class="kw">[[</span><span class="fl">1</span>]]

<span class="co"># we need genes in rows and samples in columns</span>
<span class="no">gene_expr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html">data.matrix</a></span>(<span class="no">gene_table</span>[,<span class="fl">3</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">gene_table</span>)])
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">gene_expr</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span>(<span class="st">"DATA."</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">gene_expr</span>), <span class="kw">fixed</span><span class="kw">=</span><span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gene_expr</span>) <span class="kw">&lt;-</span> <span class="no">gene_table</span>$<span class="no">GENE_SYMBOLS</span></pre></body></html></div>
</div>
<div id="running-progeny-to-get-pathway-activity-scores" class="section level3">
<h3 class="hasAnchor">
<a href="#running-progeny-to-get-pathway-activity-scores" class="anchor"></a>Running PROGENy to get pathway activity scores</h3>
<p>Activity inference is done using a weighted sum of the model genes. We can run this without worrying about the order of genes in the expression matrix using:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">progeny</span>)
<span class="no">pathways</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/progeny.html">progeny</a></span>(<span class="no">gene_expr</span>,<span class="kw">scale</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">organism</span> <span class="kw">=</span> <span class="st">"Human"</span>, <span class="kw">top</span> <span class="kw">=</span> <span class="fl">100</span>,
    <span class="kw">perm</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p>To visualize the progeny result</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">pheatmap</span>)
<span class="no">myColor</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Darkblue"</span>, <span class="st">"white"</span>,<span class="st">"red"</span>))(<span class="fl">100</span>)
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span>(<span class="no">pathways</span>,<span class="kw">fontsize</span><span class="kw">=</span><span class="fl">14</span>, <span class="kw">show_rownames</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">color</span><span class="kw">=</span><span class="no">myColor</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"PROGENy"</span>, <span class="kw">angle_col</span> <span class="kw">=</span> <span class="fl">45</span>, <span class="kw">treeheight_col</span> <span class="kw">=</span> <span class="fl">0</span>,
    <span class="kw">border_color</span> <span class="kw">=</span> <span class="fl">NA</span>)</pre></body></html></div>
<p><img src="progenyBulk_files/figure-html/unnamed-chunk-7-1.png" width="576"></p>
<p>We now have the pathway activity scores for the pathways defined in PROGENy:</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">pathways</span>)</pre></body></html></div>
<pre><code>##           Androgen       EGFR     Estrogen     Hypoxia    JAK-STAT        MAPK
## 906826  -0.4241254  0.1385179 -0.269674710 -0.06875973 -0.37598907  0.41134699
## 687983  -1.7393225 -0.9213671 -0.763490640 -1.32374232 -0.94662180 -0.90704066
## 910927  -1.5393075 -0.0695914  1.613512221 -0.76482836 -1.05428545  0.17412564
## 1240138  0.9107488 -0.2200467  0.219584368 -0.73615734 -0.07536995 -0.60238190
## 1240139 -1.0602951 -0.1846655 -0.556250681  0.13664228 -0.58644833 -0.04550095
## 906792   0.7518034  0.8045299  0.008914293  0.40118664 -0.50038060  0.77199885
##               NFkB        p53       PI3K       TGFb       TNFa      Trail
## 906826  -0.6374683 -1.0903096 -0.1953806 -0.7793323 -0.4329185 -0.8018626
## 687983  -1.3157127 -0.8692247  0.2542425 -0.8242281 -1.0195560 -0.5104074
## 910927  -0.3059911  0.8896616 -0.7487845  0.7580845 -0.1756034 -0.8092882
## 1240138  0.9755970  2.0632167  0.4839630  1.9108565  0.9865983 -0.2377919
## 1240139 -0.4751547  0.8763231  1.3814461  0.9999634 -0.3916422 -0.9224493
## 906792  -0.1878324 -0.8895494 -3.0379834 -0.3389734 -0.1436331  0.2455631
##                VEGF         WNT
## 906826  -0.13639016  0.01831199
## 687983  -0.08663407 -0.72986989
## 910927   0.13321227  2.28602257
## 1240138  0.01391272  1.51533198
## 1240139  0.90296847  1.71036886
## 906792   0.18149340 -1.22160214</code></pre>
</div>
<div id="testing-if-mapk-activity-is-significantly-associated-with-trametinib" class="section level3">
<h3 class="hasAnchor">
<a href="#testing-if-mapk-activity-is-significantly-associated-with-trametinib" class="anchor"></a>Testing if MAPK activity is significantly associated with Trametinib</h3>
<p>Trametinib is a MEK inhibitor, so we would assume that cell lines that have a higher MAPK activity are more sensitive to MEK inhibition.</p>
<p>We can test this the following way:</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">cell_lines</span> <span class="kw">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">intersect</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">pathways</span>), <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">drug_response</span>))
<span class="no">trametinib</span> <span class="kw">=</span> <span class="no">drug_response</span>[<span class="no">cell_lines</span>, <span class="st">"Trametinib"</span>]

<span class="no">mapk</span> <span class="kw">=</span> <span class="no">pathways</span>[<span class="no">cell_lines</span>, <span class="st">"MAPK"</span>]

<span class="no">associations</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span>(<span class="no">trametinib</span> ~ <span class="no">mapk</span>)
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="no">associations</span>)</pre></body></html></div>
<pre><code>## 
## Call:
## lm(formula = trametinib ~ mapk)
## 
## Residuals:
##    Min     1Q Median     3Q    Max 
## -6.548 -1.255  0.338  1.338  6.819 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) -0.94703    0.06442  -14.70   &lt;2e-16 ***
## mapk        -1.35978    0.06449  -21.09   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.998 on 960 degrees of freedom
## Multiple R-squared:  0.3165, Adjusted R-squared:  0.3158 
## F-statistic: 444.6 on 1 and 960 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>And indeed we find that MAPK activity is strongly associated with sensitivity to Trametinib: the <code>Pr(&gt;|t|)</code> is much smaller than the conventional threshold of <code>0.05</code>.</p>
<p>The intercept is significant as well, but we’re not really interested if the mean drug response is above or below <code>0</code> in this case.</p>
<p>Note, however, that we tested all cell lines at once and did not adjust for the effect different tissues may have.</p>
</div>
</div>
<div id="r-version-information" class="section level2">
<h2 class="hasAnchor">
<a href="#r-version-information" class="anchor"></a>R version information</h2>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] pheatmap_1.0.12             BiocFileCache_1.12.1       
##  [3] dbplyr_1.4.4                dplyr_1.0.2                
##  [5] progeny_1.11.2              biomaRt_2.44.1             
##  [7] DESeq2_1.28.1               airway_1.8.0               
##  [9] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
## [11] matrixStats_0.56.0          Biobase_2.48.0             
## [13] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [15] IRanges_2.22.2              S4Vectors_0.26.1           
## [17] BiocGenerics_0.34.0         knitr_1.29                 
## [19] BiocStyle_2.16.0           
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-6           fs_1.5.0               bit64_4.0.5           
##  [4] RColorBrewer_1.1-2     progress_1.2.2         httr_1.4.2            
##  [7] rprojroot_1.3-2        tools_4.0.2            backports_1.1.9       
## [10] utf8_1.1.4             R6_2.4.1               DBI_1.1.0             
## [13] colorspace_1.4-1       gridExtra_2.3          tidyselect_1.1.0      
## [16] prettyunits_1.1.1      curl_4.3               bit_4.0.4             
## [19] compiler_4.0.2         cli_2.0.2              desc_1.2.0            
## [22] bookdown_0.20          scales_1.1.1           readr_1.3.1           
## [25] genefilter_1.70.0      askpass_1.1            rappdirs_0.3.1        
## [28] pkgdown_1.5.1          stringr_1.4.0          digest_0.6.25         
## [31] rmarkdown_2.3          XVector_0.28.0         pkgconfig_2.0.3       
## [34] htmltools_0.5.0        readxl_1.3.1           rlang_0.4.7           
## [37] rstudioapi_0.11        RSQLite_2.2.0          generics_0.0.2        
## [40] BiocParallel_1.22.0    RCurl_1.98-1.2         magrittr_1.5          
## [43] GenomeInfoDbData_1.2.3 Matrix_1.2-18          fansi_0.4.1           
## [46] Rcpp_1.0.5             munsell_0.5.0          lifecycle_0.2.0       
## [49] stringi_1.4.6          yaml_2.2.1             MASS_7.3-52           
## [52] zlibbioc_1.34.0        grid_4.0.2             blob_1.2.1            
## [55] ggrepel_0.8.2          crayon_1.3.4           lattice_0.20-41       
## [58] splines_4.0.2          annotate_1.66.0        hms_0.5.3             
## [61] locfit_1.5-9.4         pillar_1.4.6           geneplotter_1.66.0    
## [64] XML_3.99-0.5           glue_1.4.2             evaluate_0.14         
## [67] BiocManager_1.30.10    vctrs_0.3.4            cellranger_1.1.0      
## [70] tidyr_1.1.2            gtable_0.3.0           openssl_1.4.2         
## [73] purrr_0.3.4            assertthat_0.2.1       ggplot2_3.3.2         
## [76] xfun_0.16              broom_0.7.0            xtable_1.8-4          
## [79] survival_3.1-12        tibble_3.0.3           AnnotationDbi_1.50.3  
## [82] memoise_1.1.0          ellipsis_0.3.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Michael Schubert, Alberto Valdeolivas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
